src/audio/google_rtc_audio_processing.c

```diff
@@ -34,8 +34,19 @@
 
 #include <google_rtc_audio_processing.h>
 #include <google_rtc_audio_processing_platform.h>
+#include <google_rtc_audio_processing_sof_message_reader.h>
 
-#define GOOGLE_RTC_AUDIO_PROCESSING_SAMPLERATE 48000
+#define GOOGLE_RTC_AUDIO_PROCESSING_SAMPLE_RATE_HZ 48000
+#define GOOGLE_RTC_AUDIO_PROCESSING_SAMPLES_PER_CHANNEL (GOOGLE_RTC_AUDIO_PROCESSING_SAMPLE_RATE_HZ / 100)
+#define GOOGLE_RTC_AUDIO_PROCESSING_DEFAULT_NUM_AEC_REFERENCE_CHANNELS 2
+#define GOOGLE_RTC_AUDIO_PROCESSING_DEFAULT_NUM_CAPTURE_INPUT_CHANNELS 1
```

Comment:
How hard will it be to make this dynamic?

Reply:
It is doable, but there are some requirements involved.

The trickiest part is that these needs to be available at _create, which means that when they are needed, there is yet no way to pass values from outside of the firmware.

Reply:
Got it, in the meantime how about kconfig? We should probably also have checks to make sure the pipeline isn't outside out spec